src/nodes/typeGuards/simple.ts

```diff
@@ -0,0 +1,247 @@
+// Code largely based on https://github.com/ajafff/tsutils
+// Original license: https://github.com/ajafff/tsutils/blob/26b195358ec36d59f00333115aa3ffd9611ca78b/LICENSE
+
+import * as ts from "typescript"
+
+export function isAbstractKeyword(node: ts.Node): node is ts.AbstractKeyword { return node.kind === ts.SyntaxKind.AbstractKeyword; }
+export function isAccessorKeyword(node: ts.Node): node is ts.AccessorKeyword { return node.kind === ts.SyntaxKind.AccessorKeyword; }
+export function isArrayBindingPattern(node: ts.Node): node is ts.ArrayBindingPattern { return node.kind === ts.SyntaxKind.ArrayBindingPattern; }
```

Comment:
A lot of these functions are already exported by TypeScript. I don't think there's a need for this library to redo the work TypeScript already does. Let's remove them.

<details>
<summary>List of existing public TypeScript APIs duplicated here</summary>

- `isArrayBindingPattern`
- `isArrayLiteralExpression`
- `isArrayTypeNode`
- `isArrowFunction`
- `isAsExpression`
- `isAssertClause`
- `isAssertEntry`
- `isAsteriskToken`
- `isAwaitExpression`
- `isBigIntLiteral`
- `isBinaryExpression`
- `isBindingElement`
- `isBlock`
- `isBreakStatement`
- `isBundle`
- `isCallExpression`
- `isCallSignatureDeclaration`
- `isCaseBlock`
- `isCaseClause`
- `isCatchClause`
- `isClassDeclaration`
- `isClassExpression`
- `isClassStaticBlockDeclaration`
- `isCommaListExpression`
- `isComputedPropertyName`
- `isConditionalExpression`
- `isConditionalTypeNode`
- `isConstructorDeclaration`
- `isConstructorTypeNode`
- `isContinueStatement`
- `isDebuggerStatement`
- `isDecorator`
- `isDefaultClause`
- `isDeleteExpression`
- `isDoStatement`
- `isDotDotDotToken`
- `isElementAccessExpression`
- `isEmptyStatement`
- `isEnumDeclaration`
- `isEnumMember`
- `isExportAssignment`
- `isExportDeclaration`
- `isExportSpecifier`
- `isExpressionStatement`
- `isExpressionWithTypeArguments`
- `isExternalModuleReference`
- `isForInStatement`
- `isForOfStatement`
- `isForStatement`
- `isFunctionDeclaration`
- `isFunctionExpression`
- `isFunctionTypeNode`
- `isGetAccessorDeclaration`
- `isHeritageClause`
- `isIdentifier`
- `isIfStatement`
- `isImportClause`
- `isImportDeclaration`
- `isImportEqualsDeclaration`
- `isImportSpecifier`
- `isImportTypeNode`
- `isIndexedAccessTypeNode`
- `isIndexSignatureDeclaration`
- `isInferTypeNode`
- `isInterfaceDeclaration`
- `isIntersectionTypeNode`
- `isJSDoc`
- `isJSDocAllType`
- `isJSDocAugmentsTag`
- `isJSDocAuthorTag`
- `isJSDocCallbackTag`
- `isJSDocClassTag`
- `isJSDocDeprecatedTag`
- `isJSDocEnumTag`
- `isJSDocFunctionType`
- `isJSDocImplementsTag`
- `isJSDocLink`
- `isJSDocLinkCode`
- `isJSDocLinkPlain`
- `isJSDocMemberName`
- `isJSDocNamepathType`
- `isJSDocNameReference`
- `isJSDocNonNullableType`
- `isJSDocNullableType`
- `isJSDocOptionalType`
- `isJSDocOverrideTag`
- `isJSDocParameterTag`
- `isJSDocPrivateTag`
- `isJSDocPropertyTag`
- `isJSDocProtectedTag`
- `isJSDocPublicTag`
- `isJSDocReadonlyTag`
- `isJSDocReturnTag`
- `isJSDocSeeTag`
- `isJSDocSignature`
- `isJSDocTemplateTag`
- `isJSDocThisTag`
- `isJSDocTypedefTag`
- `isJSDocTypeExpression`
- `isJSDocTypeLiteral`
- `isJSDocTypeTag`
- `isJSDocUnknownTag`
- `isJSDocUnknownType`
- `isJSDocVariadicType`
- `isJsxAttribute`
- `isJsxAttributes`
- `isJsxClosingElement`
- `isJsxClosingFragment`
- `isJsxElement`
- `isJsxExpression`
- `isJsxFragment`
- `isJsxOpeningElement`
- `isJsxOpeningFragment`
- `isJsxSelfClosingElement`
- `isJsxSpreadAttribute`
- `isJsxText`
- `isLabeledStatement`
- `isLiteralTypeNode`
- `isMappedTypeNode`
- `isMetaProperty`
- `isMethodDeclaration`
- `isMethodSignature`
- `isMinusToken`
- `isMissingDeclaration`
- `isModuleBlock`
- `isModuleDeclaration`
- `isNamedExports`
- `isNamedImports`
- `isNamedTupleMember`
- `isNamespaceExport`
- `isNamespaceExportDeclaration`
- `isNamespaceImport`
- `isNewExpression`
- `isNonNullExpression`
- `isNotEmittedStatement`
- `isNumericLiteral`
- `isObjectBindingPattern`
- `isObjectLiteralExpression`
- `isOmittedExpression`
- `isOptionalTypeNode`
- `isParenthesizedExpression`
- `isParenthesizedTypeNode`
- `isPartiallyEmittedExpression`
- `isPlusToken`
- `isPostfixUnaryExpression`
- `isPrefixUnaryExpression`
- `isPrivateIdentifier`
- `isPropertyAccessExpression`
- `isPropertyAssignment`
- `isPropertyDeclaration`
- `isPropertySignature`
- `isQualifiedName`
- `isRegularExpressionLiteral`
- `isRestTypeNode`
- `isReturnStatement`
- `isSatisfiesExpression`
- `isSemicolonClassElement`
- `isSetAccessorDeclaration`
- `isShorthandPropertyAssignment`
- `isSourceFile`
- `isSpreadAssignment`
- `isSpreadElement`
- `isStringLiteral`
- `isSwitchStatement`
- `isSyntheticExpression`
- `isTaggedTemplateExpression`
- `isTemplateExpression`
- `isTemplateHead`
- `isTemplateLiteralTypeNode`
- `isTemplateLiteralTypeSpan`
- `isTemplateMiddle`
- `isTemplateSpan`
- `isTemplateTail`
- `isThisTypeNode`
- `isThrowStatement`
- `isTryStatement`
- `isTupleTypeNode`
- `isTypeAliasDeclaration`
- `isTypeAssertion`
- `isTypeLiteralNode`
- `isTypeOfExpression`
- `isTypeOperatorNode`
- `isTypeParameterDeclaration`
- `isTypePredicateNode`
- `isTypeQueryNode`
- `isTypeReferenceNode`
- `isUnionTypeNode`
- `isUnparsedPrepend`
- `isUnparsedSource`
- `isUnparsedTextLike`
- `isVariableDeclaration`
- `isVariableDeclarationList`
- `isVariableStatement`
- `isVoidExpression`
- `isWhileStatement`
- `isWithStatement`
- `isYieldExpression`

</summary>

I haven't checked the other files but would want the same removals there too.

Reply:
I think it would be more user-friendly if users could just import all the type guards from this library rather than having to know which one is in this lib vs in TypeScript itself.

Maybe we could just re-export the functions:

```ts
export { isArrayBindingPattern, isArrayLiteralExpression, ... } from "typescript";
```

Reply:
Change made to re-export the functions.

Reply:
Hmmmm, I'm really not a fan of re-exporting. I can see the benefit of always having them come from the right place - but to me that doesn't outweigh the downsides of unnecessary code & having two options to take them from.

How about we file these as a followup issue and discuss there? This feels like something I'd want more eyes from the general TS community on. If I'm in the minority here we can certainly add them back in.